Standard.AI.OpenAI.Tests.Unit/Services/Foundations/AIFiles/AIFileServiceTests.cs

```diff
@@ -0,0 +1,121 @@
+                dateTimeBroker: this.dateTimeBrokerMock.Object);
+        }
+
+        public static TheoryData UnauthorizedExceptions()
+        {
+            return new TheoryData<HttpResponseException>
+            {
+                new HttpResponseUnauthorizedException(),
+                new HttpResponseForbiddenException()
+            };
+        }
+
+        public dynamic CreateRandomFileProperties()
+        {
+            int randomCreated = GetRandomNumber();
+            DateTimeOffset randomCreatedDate = CreateRandomDateTimeOffset();
+
+            dynamic randomFileProperties = CreateRandomFileProperties(
+                created: randomCreated,
+                createdDate: randomCreatedDate);
+
+            return randomFileProperties;
+        }
+
+        public dynamic CreateRandomFileProperties(int created, DateTimeOffset createdDate)
+        {
+            Stream randomStream = CreateRandomStream();
+            string randomFileName = CreateRandomString();
+            string randomFileType = CreateRandomString();
+            int randomBytesSize = GetRandomNumber();
+
+            return new
+            {
+                ExternalFile = randomStream,
+                Content = randomStream,
+                FileName = randomFileName,
+                Name = randomFileName,
+                Type = randomFileType,
+                Object = randomFileType,
+                Purpose = CreateRandomString(),
+                Id = CreateRandomString(),
+                Size = randomBytesSize,
+                Bytes = randomBytesSize,
+                Created = created,
+                CreatedDate = createdDate,
+                Deleted = GetRandomBoolean()
+            };
+        }
+
+        private Expression<Func<ExternalAIFileRequest, bool>> SameExternalAIFileRequestAs(
+            ExternalAIFileRequest expectedExternalAIFileRequest)
+        {
+            return actualExternalAIFileRequest =>
+                this.compareLogic.Compare(
+                    expectedExternalAIFileRequest,
+                    actualExternalAIFileRequest)
+                        .AreEqual;
+        }
+
+        private static bool GetRandomBoolean() =>
+            Randomizer<bool>.Create();
+
+        private static DateTimeOffset CreateRandomDateTimeOffset() =>
+            new DateTimeRange(earliestDate: new DateTime()).GetValue();
+
+        private static string CreateRandomString() =>
+            new MnemonicString().GetValue();
+
+        private static int GetRandomNumber() =>
+            new IntRange(min: 2, max: 10).GetValue();
+
+        private Stream CreateRandomStream()
+        {
+            var mockStream = new Mock<MemoryStream>();
+
+            mockStream.SetupGet(stream =>
+                stream.ReadTimeout)
+                    .Returns(0);
+
+            mockStream.SetupGet(stream =>
+                stream.WriteTimeout)
+                    .Returns(0);
+
+            return mockStream.Object;
+        }
```

Comment:
```suggestion
        private Stream CreateRandomStream() =>
            new Mock<MemoryStream>().Object;
```

`SetupGet` does nothing in this case.
There is also nothing random about this Stream